### ISSUE TITLE
Supreme Commander: Forged Alliance (9420)

### ISSUE THREAD

**Issue by gordan-bobic** [ValveSoftware/Proton#688]
Works with WINE and has done for years. Doesn't start with Proton (blank screen).

**Comment by TheChosenEvilOne** [comment]
it does start but the intro cutscene is very slow and makes it to seem like its just a blank screen.
EDIT: it seems to be caused by esync, disabled in user_settings.py and works fine

**Comment by kisak-valve** [comment]
Hello @gordan-bobic, please copy your system information from steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by admalledd** [comment]
Here is mine: https://gist.github.com/admalledd/c0e104947351c86c351e3b535e6f674d

And I confirm, disabling esync per @TheChosenEvilOne makes it work.

PS: I have to do `/tmp/proton_run winecfg` and select the "Automatically capture mouse" else it escapes to my other monitors.

**Comment by OlliC** [comment]
Can also confirm its a esync issue. But i just disabled it for this game with the launch option 

`PROTON_NO_ESYNC=1 %command%`

Probably better than disable esync for all games.

My Steam info: https://gist.github.com/OlliC/4cfd76dfa291bb19f55f

**Comment by gordan-bobic** [comment]
Confirmed that the blank screen is cured by PROTON_NO_ESYNC=1.

On to other problems

1) Not sure yet if this is SC:FA specific, but while the game is running, Steam is unaware that the game is running, e.g. if you Alt-Tab out of the game, you can start it again. This doesn't happen with native games.

2) Game is unresponsive to Alt-Enter and Alt-F4. This works on Windows, not sure if it works on standard WINE.

3) Setting Anti-aliasing too high (above 4), e.g. 8Q or 16 at very high resolution (5K) results in black screen with only a functioning mouse cursor.

**Comment by kisak-valve** [comment]
Supreme Commander Forged Alliance

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1487.
@tatsujb posted on 2018-09-16T14:04:32:

# Whitelist Request
- Name of the game to be whitelisted: Supreme Commander Forged Alliance
- Steam AppID of the game: 9420

## System Information
- GPU: GTX 760
- Driver/LLVM version: nvidia 390.48  (I did not update to 396.54 but I suspect that it would also work)
- Kernel version: 4.15.0-34-generic
- Link to full system information: https://gist.github.com/tatsujb/275e7e3817dac46c7b1bce26b1e94c15
- Proton version: 3.7

## I confirm:
- [X] that pressing the `Play` button in the Steam client is sufficient.
- [X] that [runtime config options](https://github.com/ValveSoftware/Proton#runtime-config-options)
      are necessary to run the game.
- [X] that no workarounds other than the mentioned ones are necessary.

## Issues
- [ ] I haven't experienced any issues.
- [ ] There are no issues left open for this game.
- [X] Although I consider the gaming experience equal to Windows there are
      remaining issues:

The game runs perfect past the main menu. 

(normally the games starts up to a black screen (you hear the intro video's audio but the screen is black))

I used to use `alt`-`F4` to trick the graphics stack into refreshing by making the "do you really want to exit prompt" modal show up (which I then closed).

but thanks to this github's config section : https://github.com/ValveSoftware/Proton#runtime-config-options , I've figured out that actually, enabling `PROTON_NO_ESYNC` solves this in much cleaner fashion and utterly removes menu issues altogether and makes the game as a whole run identically to windows.

Off-Topic : 
--------------

Is there a place I can get counsel on figuring out how to execute a game via proton but only in command-line?

Supreme Commander Forged Alliance's online community, despite using steam's Supreme Commander Forged Alliance is practically entirely on an online platform built around this game called Forged Alliance Forever, (aka FAF) which runs natively in Linux, in which case in a wine scenario, it does nothing specific, you simply point FAF to the bin directory in the same fashion you would under windows and the communication between the two apps happens flawlessly.

since under ubuntu 18.04.1 wine has ceased to work for a number of apps including Forged Alliance and since proton performs much better, doing all this with proton instead would be much better.

I hoped that running the app with the following commands was what steam did :

    cd ~/.local/share/Steam/steamapps/common/Proton*
    STEAM_COMPAT_DATA_PATH=/home/t/.local/share/Steam/steamapps/compatdata/9420 ./proton run /home/t/.local/share/Steam/steamapps/common/Supreme\ Commander\\ Forged\\ Alliance/bin/SupremeCommander.exe

but as it happens I must be missing some arguments or what steam does is completely different because this ends immediately with a popup error saying with `PROTON_NO_ESYNC` off : 

> Application load error P : 0000065615

or with `PROTON_NO_ESYNC` on : 

> Application load error P : 0000065434

whereas, via steam, it runs flawlessly

**Comment by fistons** [comment]
One of the nicest feature of the game is that it handles dual screen: you can have a strategic view of a screen, and a tactic view on another.

It looks like we lose this feature with Proton :cry:

**Comment by tatsujb** [comment]
@fistons we never had this feture under linux. proton is just Wine under the hood and wine itself doesn't have dual screen capabilities. 

it's unfortunate, but it's Gaz Powered Game's fault (and non-fault) at the time they were making supreme commander "fake fullscreen" A.K.A. "borderless fullscreen" didn't exist.

when it finally came out it was just like "hey look, now that scaling is made easier and that graphics cards and screens have been communicating in much simpler ways we can just maximize a window to take every pixel of the screen, no need to change resolution uselessly to what it already was and better response time for switching back to the desktop at no cost"

and it was stupid simple and nobody could believe we hadn't thought of it before. 

and implementations of dual screen that were costly cluncky and **always** had some bugs were now easy, rudimentary, reliable, robust and future proof.

the thing is, under windows most people use a hacked-in fake fullscreen anyways, I even made a script myself : https://forums.faforever.com/viewtopic.php?f=2&t=9778

the thing is you'll have to hack proton into "fake dualscreening" also. I've tried with wine, and never really got any success. 

hopefully it might work this time around with proton.

**Comment by pipnina** [comment]
I've noticed that this game, when run through Proton or naked Wine, is liable to a crash during gameplay where the game completely freezes and needs to be killed.
I have my game log attached, near the end you can see where the error occurs (err:ntdll:RtlpWaitForCriticalSection)
[steam-9420.log](https://github.com/ValveSoftware/Proton/files/2677581/steam-9420.log)

If not killed, that error shows up every minute or so (it retries).

EDIT: This can be most easily replicated by using an 81km map, with maximum number of AI. Takes between 5 and 20 minutes to produce.

**Comment by douglasg14b** [comment]
Is there a way to launch the game through proton via command line?

The community lobby, mod manager, and re-balances/fixes seems to need to operate the game through the command line which seems to be problematic for proton.

https://forums.faforever.com/viewtopic.php?f=2&t=4507&start=620#p170186

**Comment by tatsujb** [comment]
@douglasg14b hey, good to see I'm not the only one asking for this, I asked this in this very thread : https://github.com/ValveSoftware/Proton/issues/688#issuecomment-421791298
hopefully something happens with this soon.

**Comment by douglasg14b** [comment]
@tatsujb Are there any specific persons that are heavily involved in the proton project that we might be able to ping to get them to chime in on that?

**Comment by pipnina** [comment]
There is someone with "valve" in their name subscribed to this issue. They might chime in on this when they get to that point in their notifications/emails. I imagine with a piece of software that has 1500+ issues the valve employees get quite a backlog.

---

On note: I also would quite like to see FAF runnable with Steam/Proton Supreme Commander.

**Comment by tatsujb** [comment]
@pipnina yeah it's possible the bug you mentioned : https://github.com/ValveSoftware/Proton/issues/688#issuecomment-447109059 isn't even reproducible when running with FAF.
FAF fixed so many FA bugs.

**Comment by OlliC** [comment]
> @tatsujb Are there any specific persons that are heavily involved in the proton project that we might be able to ping to get them to chime in on that?

That would be @aeikum.

**Comment by aeikum** [comment]
> Is there a way to launch the game through proton via command line?
> 
> The community lobby, mod manager, and re-balances/fixes seems to need to operate the game through the command line which seems to be problematic for proton.
> 
> https://forums.faforever.com/viewtopic.php?f=2&t=4507&start=620#p170186

Depending on exactly what you need, you can use the PROTON_DUMP_DEBUG_COMMANDS thing to generate `/tmp/proton_$USER/run`. Then you can run `/tmp/proton_$USER/run SomeExecutable.exe -WithParameters`.

**Comment by tatsujb** [comment]
translation : open up file folder, search :  `user_settings.sample.py` , rename to : ` user_settings.py`, open with editor,  uncomment : `"PROTON_NO_ESYNC": "1",`, add `"PROTON_DUMP_DEBUG_COMMANDS": "1",`
I'm still downloading FA. about to give it a spin.

@aeikum thanks a lot for this, you seemed to ask what we needed specifically.

well the first real issue is running FA with proton-steam from command line, this fails with errors ranging from `Application load error P : 0000065615` to `Application load error P : 0000065434`

have you successfully run this game with command line? for me it only works from steam gui.

EDIT 1 : 

huh, hadn't tried this in awhile but today I can't run it at all : 

![screenshot from 2019-01-24 21-39-35](https://user-images.githubusercontent.com/5132359/51707120-aae9c480-2020-11e9-91b2-766c5eda8d06.png)


EDIT 2 :

okay so editing user_settings.py while the game is downloading is not a good idea because it'll be in the downloading folder while the game is downloading and only when it's done, it'll be moved to the proton folder.

So this time I waited untill the game had finished downloading to find and edit that file

I also forced the proton version to : 3.16-4

and now it runs. 
I'm going to try running it with command line now....

EDIT 3  : 

ok so that's been several months coming, it works, omg aeikum where have you been?

for me it was :
`/tmp/proton_t/run /home/t/.steam/steam/steamapps/common/Supreme\ Commander\ Forged\ Alliance/bin/SupremeCommander.exe`

trying it in FAF now.

EDIT  4 : 

omg this is everything I've striven for for months!!

it works !!!

the following settings will get you FAF + FA native with no wine needed !! (minus the "technically..." )

(FAF settings don't want the backslashes, I think it has to do with JAVA)
![screenshot from 2019-01-24 22-21-48](https://user-images.githubusercontent.com/5132359/51709463-b4762b00-2026-11e9-8959-0281c1ffd064.png)

**Comment by douglasg14b** [comment]
@tatsujb  Can you write-up a summarized step-by step that we could copy/paste here and there? It's amazing that you did the ground work for this.

Does this actually work completely with FAF as if it where installed and configured under Wine? I so, this has made my week. And should be posted on the Linux thread on the FAF forums.

**Comment by douglasg14b** [comment]
@tatsujb  There should be no need to edit the `user_settings.sample.py` file. 

Also note that the tmp folder is a poor place to keep the files since they may be removed on reboot.

You can set the launch options for the game in steam and it generates the appropriate `/tmp/proton` folder.

`PROTON_NO_ESYNC=1 PROTON_DUMP_DEBUG_COMMANDS=1 %command%`

![image](https://user-images.githubusercontent.com/1400380/51725629-4b87b680-2018-11e9-84b2-ce1c5ae2261d.png)

**Comment by tatsujb** [comment]
> @tatsujb Can you write-up a summarized step-by step that we could copy/paste here 


If you want but I'm in the process of making a youtube video-guide which in my mind is even better (yes it will include a pastebin of copy-pastable commands)

> @tatsujb There should be no need to edit the `user_settings.sample.py` file.
> You can set the launch options for the game in steam and it generates the appropriate `/tmp/proton` folder.
> 
> `PROTON_NO_ESYNC=1, PROTON_DUMP_DEBUG_COMMANDS=1 %command%`

yeah I'd forgotten about that way. I'll prefer this method in the on-screen instructions and in the attached pastebin.

> Also note that the tmp folder is a poor place to keep the files since they may be removed on reboot.

is copying the folder contents to somewhere else viable? what I mean is, will proton recognize this or does it expect tmp specifically?

**Comment by tatsujb** [comment]
yeah copying that file or the whole proton_$USER dirrectory to say, documents, for example, and pointing there instead of tmp works.

**Comment by douglasg14b** [comment]
@tatsujb  I played a couple games with this kind of install and have had issues with mods working, and explosions for munitions seems to be missing for lots of things.

Have you had any issues?

Edit: Tried more mods and none of them seem to actually load or work. The "Mods" section of game info when the game starts is empty.

**Comment by tatsujb** [comment]
@douglasg14b you could have used my old tutorial video instead of getting impatient :'(

```
cd ~/.steam/steam/steamapps/common/Supreme\ Commander\ Forged\ Alliance
ln -s ~/My\ Games/Gas\ Powered\ Games/Supreme\ Commander\ Forged\ Alliance/Maps/ Maps
ln -s ~/My\ Games/Gas\ Powered\ Games/Supreme\ Commander\ Forged\ Alliance/Mods/ Mods
/home/t/.steam/steam/steamapps/compatdata/9420/pfx/drive_c/users/steamuser
rm -rf My\ Documents
mkdir My\ Documents
cd My\ Documents
ln -s ~/My\ Games/ My\ Games
chmod +x ~/.faforever/bin/ForgedAlliance.exe
```


it's almost done uploading wait :P

**Comment by douglasg14b** [comment]
I did! Though I wasn't sure what you where doing with the linking compared to the new way, but now I see...

I haven't played FAF for years, now I use linux and had an FAF itch to scratch so I was pretty stoked that it might work. Definitely was impatient.

**Comment by tatsujb** [comment]
https://www.youtube.com/watch?v=4B6EsXtL8C0

**Comment by douglasg14b** [comment]
@tatsujb Thanks! Though not that I do not have a `/285330/` dir under `steamapps/compatdata/` FYI.

In my case it was `9420`. This can be found by going to `$HOME/.steam/steam/steamapps/common/Supreme Commander Forged Alliance/` and opening the `installscript.vdf` file, the app ID will be at the end of `HasRunKey`. For example: `"HasRunKey"     "HKEY_LOCAL_MACHINE\\Software\\Valve\\Steam\\Apps\\9420"`

**Comment by tatsujb** [comment]
my bad, you were right it should have been : 

```
/home/t/.steam/steam/steamapps/compatdata/9420/pfx/drive_c/users/steamuser
rm -rf My\ Documents
mkdir My\ Documents
cd My\ Documents
```

that works for me.

**Comment by douglasg14b** [comment]
@tatsujb 

Any experiance with this issue wehn trying to do ranked 1v1?

```
Failed to setup initial search path.

Program : Z:\home\douglas\.faforever\bin\ForgedAlliance.exe
Cmd line arguments : /init init.lua /nobugreport /aeon /log /home/douglas/.faforever/logs/game.log /gpgnet 127.0.0.1:33887 /mean 814.79 /deviation 203.135 /savereplay gpgnet://127.0.0.1:39101/9211177/douglasg14b.SCFAreplay /country US /numgames 7 /team 1 /players 2

Callstack:
    unavailable.

Last 100 lines of log...

warning: Running "Z:\home\douglas\.faforever\bin\init.lua" failed: cannot read init_faf.lua: No such file or directory
         stack traceback:
         	[C]: in function `dofile'
         	Z:\home\douglas\.faforever\bin\init.lua(1): in main chunk
```

Edit: I see the thread here https://github.com/FAForever/downlords-faf-client/issues/1095

Commented there.

**Comment by douglasg14b** [comment]
@tatsujb Do you see issues with the Steam FAF install where there are no explosions? Artillery, bombers...etc nothing makes any explosions.

**Comment by tatsujb** [comment]
ok are you ATi? and check your graphics settings in game (maxing them out helps) but this is graphics and driver related. FA plays better with nvidia and doesn't require much GPU. if you don't mind switching out graphics from time to time, even an old shitty nvidia will get you better results than ati.

but it will never be perfect.

wine has never (and probably will never) support bloom. FA's tech for spots of light in bright areas. making nukes shine like the sun and ect.

I don't see a lot of projectiles, in particular, those of t1 point defense and of the commander, especially with Aeon.

I rely on strategic dots of the projectiles when zoomed out alot.

I dunno what to say... :/

**Comment by douglasg14b** [comment]
Hu, yeah. I am using ATI, and RX 580. That's pretty weird if that's the problem.

I have a gtx 960 laying around, wonder if Ubuntu can handle two graphics cards, and if supcom can use tone over the other. I'll have to try.

It's not vital information, just nice to have the visual when doing things like bombing an ACU.

**Comment by kisak-valve** [comment]
Hello @douglasg14b, please copy your system information from steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by douglasg14b** [comment]
```
OS: Ubuntu 18.10
KERNEL: 4.18.0-12-generic
CPU: AMD Ryzen 7 2700X Eight-Core
GPU: Radeon RX 580
GPU DRIVER: 4.4 Mesa 18.2.2
RAM: 32 GB
```

<details>
<summary>Detailed Info</summary>

```
Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD Ryzen 7 2700X Eight-Core Processor         
    CPU Family:  0x17
    CPU Model:  0x8
    CPU Stepping:  0x2
    CPU Type:  0x0
    Speed:  4000 Mhz
    16 logical processors
    8 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Ubuntu 18.10 (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.18.0-12-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12001000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime-beta-release_2018-11-23

Video Card:
    Driver:  X.Org Radeon RX 580 Series (POLARIS10, DRM 3.26.0, 4.18.0-12-generic, LLVM 7.0.0)
    Driver Version:  4.4 (Compatibility Profile) Mesa 18.2.2
    OpenGL Version: 4.4
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 143 Hz
    VendorID:  0x1002
    DeviceID:  0x67df
    Revision Not Detected
    Number of Monitors:  3
    Number of Logical Video Cards:  1
    Primary Display Resolution:  2560 x 1440
    Desktop Resolution: 5560 x 1920
    Primary Display Size: 27.44" x 15.43" (31.46" diag)
                                            69.7cm x 39.2cm (79.9cm diag)
    Primary VRAM: 8192 MB

Sound card:
    Audio device: ATI R6xx HDMI

Memory:
    RAM:  32151 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  921682 Mb
    Largest Free Hard Disk Block:  654525 Mb
    VR Headset: None detected

Recent Failure Reports:

```
</details>

**Comment by tatsujb** [comment]
I don't know if this is pertinent but apparently people have messed with having both an ati and nvidia graphic card onboard at the same time on ubuntu : https://streamhpc.com/blog/2011-10-12/both-nvidia-gtx-and-amd-radeon-on-linux/

i personally had both types of graphic cards plugged in at one point but I was blacklisting the nvidia to do hardware passthrough to emulated windows.

if worse comes to worse it doesn't hurt to have both in at the same time and switch out graphics drivers with a reboot to utilize one then the other.

**Comment by foodlfg** [comment]
> I've noticed that this game, when run through Proton or naked Wine, is liable to a crash during gameplay where the game completely freezes and needs to be killed.
> I have my game log attached, near the end you can see where the error occurs (err:ntdll:RtlpWaitForCriticalSection)
> [steam-9420.log](https://github.com/ValveSoftware/Proton/files/2677581/steam-9420.log)
> 
> If not killed, that error shows up every minute or so (it retries).
> 
> EDIT: This can be most easily replicated by using an 81km map, with maximum number of AI. Takes between 5 and 20 minutes to produce.

i can confirm this.
tldr; i have AMD GPU and i get these crashes constantly. i've tested with Nvidia GPU on the same hardware/software config and the crashes disappeared. both vanilla SupComFA and modded SupComFAF has this problem under Wine or under Proton Wine (version doesn't matter)


-----
i did extensive testing regarding this:
different Ubuntu releases, kernel versions, Wine versions (v2.0 - v3.18), Wine settings, game settings, game scenarios, open source AMD graphics drivers and the closed ones downloaded from AMD's site. 
Non of this solved the crashes.

what i've found is that the games crash at the same "game time" at ~40-50 mins (game time is simulation speed dependent). in real world time it's between 20 mins - 40 mins.

Easiest way to test this in vanilla SupComFA:
map: Betrayal Ocean, 4AI-s, game type: sandbox, game speed: fast

tested configs:
AMD Ryzen5 1600X, 16GB RAM, AMD RX560     4GB RAM - crashes at ~48min game time
AMD Ryzen5 1600X, 16GB RAM, Nvidia GT710  2GB RAM - no crashes

AMD drivers: the open source and the closed source provided by AMD
Nvidia drivers: open source Nouveau driver and the proprietary driver provided by Ubuntu

Steam info
https://gist.github.com/foodlfg/ba0cb22830aeff726c9dffc5c7d4dc2d#file-rx560-opensource-steam-info-txt

**EDIT / UPDATE:**
the crashes appears on Nvidia too for me. 
see below: https://github.com/ValveSoftware/Proton/issues/688#issuecomment-460047828

**Comment by foodlfg** [comment]
> Hu, yeah. I am using ATI, and RX 580. That's pretty weird if that's the problem.
> 
> I have a gtx 960 laying around, wonder if Ubuntu can handle two graphics cards, and if supcom can use tone over the other. I'll have to try.
> 
> It's not vital information, just nice to have the visual when doing things like bombing an ACU.

@douglasg14b 
could you do tests for my problem (seen above) using your AMD and Nvidia card?
remove your AMD card then test with Nvidia, no need for messy dual GPU configs.

Thanks!

**Comment by kisak-valve** [comment]
~Hello @foodlfg, can you test with a newer version of mesa like from a PPA (Padoka, Padoka stable, Oibaf)? If they are also affected, it may be worthwhile to also bring your issue to the attention of [your video driver vendor](https://www.mesa3d.org/bugs.html).~

**Comment by pipnina** [comment]
> i've tested with Nvidia GPU on the same hardware/software config and the crashes disappeared.

Interesting, I only have Nvidia cards but the crash occurs for me. What driver version do you use for your Nvidia cards? I'm currently using 396.54

**Comment by tatsujb** [comment]
415 for me.

**Comment by foodlfg** [comment]
> > i've tested with Nvidia GPU on the same hardware/software config and the crashes disappeared.
> 
> Interesting, I only have Nvidia cards but the crash occurs for me. What driver version do you use for your Nvidia cards? I'm currently using 396.54

You're right unfortunately.
I installed my GT710 (proprietary drives) and tested again using different replays (i did the Betrayal Ocean gameplay test too) and it crashed like it does with AMD.
When I tested the GT710 before I probably chose the wrong replays (they wasn't long enough). Sorry about the confusion.

I couldn't test with Proton Wine and vanilla SupcomFA (the Launch button didn't appear in Steam...) but that doesn't really matter. It definitely has the same "late game" crashes (on my system on linux).

Now I have no idea what causes the crashes. Maybe the game engine does something in late game (30+ mins in) that Wine cannot handle. I don't know.

The crash log is the same btw:
```
EXCEPTION_ACCESS_VIOLATION (0xc0000005) at address 0x0095854f
    attempted to write memory at 0x01f60000

Program : Z:\home\test1\FAForever\bin\ForgedAlliance.exe
Cmd line arguments : /replay /home/test1/FAForever/cache/temp.scfareplay /init init_fafbeta.lua /nobugreport /log /home/test1/FAForever/logs/replay.log /replayid 9223888

Callstack:
	Unknown symbol (address 0x0095854f)

Last 100 lines of log...
```
The full log can be found here:
https://gist.github.com/foodlfg/ba0cb22830aeff726c9dffc5c7d4dc2d#file-full-crash-log-txt

The full glxinfo for the GT710 with proprietary drivers:
https://gist.github.com/foodlfg/ba0cb22830aeff726c9dffc5c7d4dc2d#file-gt710-proprietary-glxinfo-txt
4.6.0 NVIDIA 390.87 - i guess.

**Comment by pipnina** [comment]
Yeah. I can now confirm that neither the 396 driver or the 415 driver solve the ntdll:RtlpWaiForCriticalSection bug. I just tested the same map on both and while the 415 driver got further than the 396 driver, the crash still occurred.

**Comment by tatsujb** [comment]
can you guys give a step by step to reproduce? am I supposed to use a particular replay file?

**Comment by douglasg14b** [comment]
I'm curious about this as well, I receive no crashes even on long games with AMD.

**Comment by pipnina** [comment]
@tatsujb @douglasg14b 
I produce it by loading a skirmish 81km map like Betrayal Ocean, and filling the 7 AI slots (any difficulty). It's possible that increasing the time modifier makes the crash occur faster but it seems to happen for me at around the 20-30 minute mark on an 81km map. Happens after over an hour when it occurs in 40km maps, but it doesn't seem to happen at all in 20km or smaller. Can't get replays from it because the game crashes.

If there's a way to get more detailed information about the crash (debug tools etc) I can reproduce it easy enough to provide that info.

**Comment by douglasg14b** [comment]
So, this might be related. I cannot play games with AIs with other players. I become de-synced quickly (The desync that happens when the simulations hashsdoesn't match up between players).

Replays with AIs also playback differently for me than for how they actually went down.

**Comment by tatsujb** [comment]
>  It's possible that increasing the time modifier makes the crash occur faster 

you mean speed? can you reproduce it if you just leave it on normal?

**Comment by tatsujb** [comment]
> So, this might be related. I cannot play games with AIs with other players. I become de-synced quickly (The desync that happens when the simulations hashsdoesn't match up between players).
> 
> Replays with AIs also playback differently for me than for how they actually went down.

you were observer? did you have any other players? are your ports forwarded?

**Comment by douglasg14b** [comment]
@tatsujb 

I was a player. But watching the replay does not cause the same turn of events that actually happened in the game according to those that where in it.

The Desync isn't a network-related item, it's when the hash of the games state does not match between the clients, and the game diverges from there on the desynced client. This only happens with AI present (I'm trying to introduce a few new players to the game).

**Comment by pipnina** [comment]
@douglasg14b Are you running with mods? I have found that mods need to be installed in a rather specific way or multiplayer de-syncs will occur. BrewLAN and Experimental Wars seem to cause this a lot but I've not experienced this with a vanilla game.
I think it is also not a proton issue, since my windows friends experience modded multiplayer de-syncs as well.

@tatsujb It can be reproduced at normal game speeds, I just tested it on 81km map "Debris" at fixed standard game speed and it crashed in 16 minutes and 40 seconds.

**Comment by douglasg14b** [comment]
@pipnina UI mods only, and the issue only occurs in multiplayer games with AIs

**Comment by tatsujb** [comment]
very helpful details. did you also leave the game minimized and check on it once or twice?

I was able to reproduce the crash (not the desync) in these conditions.

I'm retesting now with all mods turned off (I use Ui party, for example)

EDIT : yeah it occurs at 24:31 even with no mods at all.

**Comment by douglasg14b** [comment]
I was also able to reproduce the crash in longer ~40min games with AI.

I've yet to have any PVP games go over 30+mins.

**Comment by tatsujb** [comment]
but did you try without ai? (i'm willing to try with you now)


these crashes may be caused by defects in FA coupled with wine.

Fa crashed this way in windows to when it was fullscreen-ed but a couple runtime bugfixes plus a switch to windows 10 made the crashes less prone to happen. 

both before and after the bugfixes, making FA run in windowed made the bugs disappear entirely.

on linux, under wine I found that running wine in emulated desktop mode had a similar positive effect. the issue is I don't know how to enable that setting under proton.

Actually, come to think of it,when I do the mick-mac mentioned here : https://github.com/FAForever/downlords-faf-client/issues/1095#issuecomment-462430344

do I even get the `PROTON_NO_ESYNC=1, PROTON_DUMP_DEBUG_COMMANDS=1 %command%` loaded?

**Comment by lemorragia** [comment]
I'm experiencing some issues too, the game starts and run, i can start a game and nothing crashes BUT the cursor keeps disappearing every time i move it(in game and inside menus). 

At the same time when i move it(or click, or using the mousewheel) there's some kind of mini-lag/little freeze in the game...this impacts the overall gameplay and performance in a way that it becomes very difficult to play (apparently if i keep the mouse still, the game seems to work without a problem).

For all those wanting to run the FAF lobby there's a video tutorial online: https://www.youtube.com/watch?v=4B6EsXtL8C0

nevertheless the game has the mouse problem with AND without the lobby.
I'm running on proton 3.16.7-beta (but the problem seems to exists with other versions too), Nvidia 930M 390.77 drivers and ubuntu 18.04.2 LTS

**Comment by foodlfg** [comment]
@lemorragia 
as for the flickering mouse issue,
disable v-sync (Vertical Sync - off) on the video tab in the settings.
this fixed it for me.

as for the other one,
i experience lags like this but only if there are many units on screen + zooming in. it happens for me when the game starts loading in detailed models and textures (LoD). for me, it's not distracting, but i have a faster graphics card (AMD RX560 with 4GB memory). 
i know no fix for that, try to test it with better hardver (preferably desktop) or play with the video settings. GeForce 930M is a GPU for laptops, probably the amount of memory it has and its memory speed are not the best.

note:
i'm running the game with _wine virtual desktop emulation ON 1080p_ (not accessible when you use proton/wine) and the _game is set to full screen 1080p_.

**Comment by lemorragia** [comment]
@foodlfg thanks.

I tried a little bit of configurations, with no effects on this problem:
- vsync on/off
- updating nvidia drivers to 390.87
- fullscreen/windowed mode
- "lock fullscreen cursor to window" option on/off
- virtual desktop emulation on/off (cloning the temporary directory on proton and running {protoncopy}/run winecfg gives you access to wine configuration for proton
- with or without FAF lobby (which works correctly, btw)

This effect manifests itself (strangely) even in the main menu(if i move the mouse a little bit, the menu itself get "stuck"), or at the very beginning of a sandbox game. It's true that my graphic card is not the best for the game, but i played it on windows several times and it was ok on small maps with limited number of units. Also the cpu is a quite performant i7, so probably the hiccup isn't there.

**Comment by foodlfg** [comment]
@lemorragia 
{protoncopy}/run winecfg
i see, thanks!

-----------------
a different topic for those who wants to test this.

i have mentioned here the memory access violation issue that causes the game to crash in late game (more info about this in this thread above). 
```
EXCEPTION_ACCESS_VIOLATION (0xc0000005) at address 0x0095854f
    attempted to write memory at 0x01f60000
```

it seems that this issue can be fixed by using Wine 4.0 with Gallium Nine enabled.
Wine 4.0 only is not enough. with Gallium Nine tho, i could play over 1 hour long games against AIs on big maps.
i haven't tested team games tho. and your experience may vary if you use Nvidia card with Gallium Nine. i haven't tested with Nvidia either. 

what I have done:
- i did not install any Mesa graphics drivers. Ubuntu 18.10 has Mesa drivers with Gallium Nine supported. also, i had stability problems with Mesa drivers from Oibaf's PPA before.
- i installed wine-d3d9 from this ppa: https://launchpad.net/~commendsarnex/+archive/ubuntu/winedri3
- this installed Wine 4.0 on my system (note, the wine version often changes in this ppa)
- i enabled Gallium Nine in the staging tab of winecfg

side effects: 
- Gallium Nine caused the map appearing all green with no units displaying.
i could fix this by closing the minimap. this problem also happens when you switch on split screen mode in the game. it looks like the game can render only one map correctly when Gallium Nine is enabled in Wine.

more testing is needed but this looks promising so far.

note:
Gallium Nine is not available in Proton/Wine. though there is an unofficial repo for that on github.

**Comment by lemorragia** [comment]
@foodlfg wow! that's very good news indeed

I gave a quick try at gallium nine + proton [guide here](https://github.com/popsUlfr/Proton/releases/tag/proton-3.16-6-gne-0.1.1) ... the game works(and everything seems more performant indeed) but the mouse problem remains. It is good to know, though, that with some configuration can work on steam too.

More infos on my problem, if anyone is experiencing something similar: 

- probably it is connected only to mouse movement...zooming in/out on the map carefully keeping the the cursor on the same place on the map avoids the problem
- interestingly, as long as you move the mouse, the game is stuck. The game time freeze and jumps forward when you stop the mouse movement...almost a "time-skipping feature" :-D

next i'm gonna try run it on a live system to exclude some strange problems with my ubuntu installation

**Comment by tatsujb** [comment]
actually @foodlfg, this is very relevant. Proton/Valve should likely be able to integrate Gallium nine as that code is open source.

I think if Valve deems it pertinent to the functioning of their games they will make the right move here.

**Comment by lemorragia** [comment]
@foodlfg did a quick try on a fresh ubuntu-on-usb. The time-skipping problem goes away, though the flickering/disappearing of the cursor remains...i'm prone to think that maybe is something with the drivers/library and my particular hardware

I also did some testing with steam+gallium nine and it seems smoother in performance (i didn't measure fps though)... @tatsujb  i'm not sure if valve is ever going to integrate it in proton, the development seems quite stuck as far as i can see

**Comment by tatsujb** [comment]
Isn't it a case of "this is a good idea! let's fork this as our own for intergration to steam!" like they did with wine?

Wine devs got tired and moved on, that doesn't mean wine was a bad idea and both valve and lutris saw that and started pumping time and effort into their own forks of it. And we see the positive effects of that today.

I can understand that gallium nine isn't quite ready yet but I don't think that's the question, the idea is undeniably good.

Anyways. thank you very much for the testing, please do keep it up. I'm very low on time right now but I'm looking to have more time this weekend. I should get back to you both on this then.

**Comment by kisak-valve** [comment]
Hello @tatsujb, @lemorragia, please use #66 to discuss gallium nine with Proton, using it is not game-specific.

**Comment by douglasg14b** [comment]
Cannot join or create any other non-FAF game types (like phantom, xtreme wars, labwars....etc) games either:

![image](https://user-images.githubusercontent.com/1400380/53701383-7d2b3480-3db1-11e9-9d8f-e9a5222dbbc2.png)

@tatsujb  have you had this issue?

```
Failed to setup initial search path.

Program : Z:\home\douglas\.faforever\bin\ForgedAlliance.exe
Cmd line arguments : /init init.lua /nobugreport /log /home/douglas/.faforever/logs/game.log /gpgnet 127.0.0.1:46577 /mean 1114.19 /deviation 162.347 /savereplay gpgnet://127.0.0.1:34001/9350837/Zongrash.SCFAreplay /country US /numgames 22

Callstack:
    unavailable.

Last 100 lines of log...

warning: Running "Z:\home\douglas\.faforever\bin\init.lua" failed: cannot read init_faf.lua: No such file or directory
         stack traceback:
         	[C]: in function `dofile'
         	Z:\home\douglas\.faforever\bin\SupComDataPath.lua(2): in main chunk
         	[C]: in function `dofile'
         	Z:\home\douglas\.faforever\bin\init.lua(46): in main chunk
```

**Comment by tatsujb** [comment]
I'm pretty sure this is caused by :

A : maps and mods folder aren't correctly soft-linked to the dir you have at `/home/USERNAME/My\ Games/Gas\ Powered\ Games/Supreme\ Commander\ Forged\ Alliance` 
B : when  "Execution Dirrectory"  is not in the `/home/USERNAME/.faforever/bin` format

(with 90% of my bet being on A, that being said 1v1 ladder doesn't work without B)

I'm working hard to get next video out : 

I've made a script that will make the whole thing more of a breeze (and once people contribute to it to make the script more robust an malleable, it will suit most needs)

but for now this script should only be tested on a freshly installed ubuntu 18+ (although the "freshly installed" is just a precaution on my part I haven't had time to test on other case figures, but logically, it should work on systems with previous installs of FA)

oh and by the way, yes this script does handle setting up gallium nine for you.

**Comment by douglasg14b** [comment]
Hm, I have the map and mods folders symlinked, I also have the execution directory laid out.

I have `/home/USERNAME/My Games/Gas Powered Games/Supreme Commander Forged Alliance/Mods` symlinked to `/home/USERNAME/.steam/steam/steamapps/common/Supreme Commander Forged Alliance/Mods/`

I did however have a snafu with my mods folder a while back where it got wiped, but I didn't think that would have an effect since the client should just re-download mods? Unless it tries to keep track of them separately, somewhere else. I don't see Phantom, Xtremewars or others in the mods folder though, even after the FAF client downloaded it, are game-types stored somewhere else?

I do get this whenever I start the FAF client, all the folders it lists are empty (the mods folder seems to contain empty directories for every subdirectory for each mod, as if the directory structure was flattened out. The original directory still cotnains all the mods files though):
![image](https://user-images.githubusercontent.com/1400380/53702638-ef097b00-3dbd-11e9-844e-e107d3dc13b8.png)

**Comment by tatsujb** [comment]
I would strongly recommend deleting .faforever and .com.faforever.client.FafClientApplication in your home directory (thereby loosing your configuration) and starting back up fresh. before you do though, read ahead an re-link maps and mods.

> I did however have a snafu with my mods folder a while back where it got wiped, but I didn't think that would have an effect since the client should just re-download mods? 

ofc. yes. emptying it out fixes it if the pointer is henceforth correct. IF. 

I'm not sure it is.

steamapps folder for steam might be named  `SteamApps` or `steamapps` which meens my previous guidlines have to be updated to take that into account. (not on older systems with the old folder name as steam keeps it if it was already like that)

look at the end of my linked script try running just those commands. modify them first to turn SteamApps back into steamapps if that is what you have.

**Comment by douglasg14b** [comment]
I'll give that a shot momentarily.

After some more investigation, the error might be from the relative file-path not working with lua? specifically `dofile('FILENAME.lua')`. All the errors for game types and for 1v1 all point to `dofile()` not being able to find the file, and the normal `faf` and `faf_beta` game types don't try to `dofile()` any files in the same directory as themselves.

I tried to modify the lua files to test some things, but the FAF client returns it to it's original state before trying to execute it (I iamgien it checks the file hash, and re-downloads it if it doesn't match)

**Comment by tatsujb** [comment]
yep it does do that.

https://github.com/tatsujb/installFAFscript

wow version 1.3 was garbage.

v 1.4 is a tad better.

**Comment by tatsujb** [comment]
guys try my install script I'm up to version 1.9 and I've had two people on voicechat with successes using it so far : https://github.com/tatsujb/installFAFscript

(hopefully your distro is ubuntu or an ubuntu flavor or mint or elementary, otherwise, I'm working on it)

**Comment by lemorragia** [comment]
@tatsujb outstanding work...thank you so much...
i plan to reinstall my system from scratch (right now my problems are somehow connected to my current installation) and i'll sure will give it a try!

**Comment by tatsujb** [comment]
@lemorragia btw that script uses gallium-proton (but you can choose to opt-out) thanks to you pointing out it's existence.

**Comment by nothimofc** [comment]
hey, uh so i'm having a issue with Supreme Commander, not Supreme Commander FA, so when launching it just stucks on black screen and i see the game cursor, also no sound, tried disabling esync and enabling debug thing but still no things, have proton 16.4

EDIT : Tried to set fidelity to 1, it regenerated the .pref file to default..

**Comment by tatsujb** [comment]
@osu you haven't read this thread : you need to right click on "Supreme Commander" in your steam library, open up "Preferences" the "SET LAUNCH OPTIONS" and type : "`PROTON_NO_ESYNC=1 %command%`"

**Comment by tatsujb** [comment]
@osu could you include what steam exports as system info (from the help menu)
also why not try with proton 3.16-8 Beta?

**Comment by tatsujb** [comment]
@lemorragia  how did it go?

**Comment by lemorragia** [comment]
@tatsujb couldn't try it yet...work issues...i'll try as soon as possible ;-)

**Comment by Sl0vi** [comment]
@osu Upgrade to Proton 4.2-1, it fixes the issue with bloom render in Supreme commander.

You still need to disable esync in your launch options.

I wrote a guide on how to get Supreme Commander to work on older versions of Proton in my bug report: #1107

**Comment by pipnina** [comment]
Seems to me that 3.16 works fine, but 4.2 causes de-syncs in multiplayer. Can anyone else confirm this?

**Comment by tatsujb** [comment]
I can : http://forums.faforever.com/viewtopic.php?f=2&t=4507&sid=6117ec11b5d786169c1c7a46854f1735&start=690#p173139

@Sl0vi @pipnina, try my script ? : https://github.com/tatsujb/installFAFscript

**Comment by tatsujb** [comment]
so in conclusion there is definitely a bug with 4.2-2 and 4.2-3 which causes users who have 4.2-x to desync in online games.
 
if you switch back to 3.16-9 beta the desyncs disappear.

should I keep it in this thread or create a new one?

**Comment by kisak-valve** [comment]
Hello @tatsujb, because this game is unofficially supported, this issue report is the right place for all game-specific issues related to this game. Please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and, reproduce the regression, and drag and drop the generated $HOME/steam-$APPID.log into the comment box. You may need to throw it in an archive if it's a large log.

**Comment by vegai** [comment]
On Vega 54, Forged Alliance seems to work perfectly (based on 2 hours of gameplay) on latest proton 3.16, but not at all on proton 4.2. On 4.2, the game apparently starts fine (sounds of the starting screens can be heard) but everything's black.

**Comment by kisak-valve** [comment]
Hello @vegai, with Proton 4.2-3, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and drag and drop the generated $HOME/steam-$APPID.log into the comment box. Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by douglasg14b** [comment]
@tatsujb  When are you available next? Lets do a quick game with 4.2 to get a desync out of it for the log file. If you have the time.

**Comment by pipnina** [comment]
I might be able to get a test done with my friends who have the game tomorrow.

**Comment by douglasg14b** [comment]
Hm, the log file is not generated unless it's ran from steam it seems. 

@kisak-valve We're running the game through the debug output script `/tmp/proton_$user/run`, is there a way to get `PROTON_LOG` to work from there?

Example:

```bash
#!/bin/bash
#Run game or given command in environment

cd "/home/douglas/.steam/steam/steamapps/common/Supreme Commander Forged Alliance"
DEF_CMD=("/home/douglas/.steam/steam/steamapps/common/Supreme Commander Forged Alliance/bin/SupremeCommander.exe")
PATH="/home/douglas/.steam/steam/steamapps/common/Proton 4.2/dist/bin/:/home/douglas/.steam/ubuntu12_32/steam-runtime/amd64/bin:/home/douglas/.steam/ubuntu12_32/steam-runtime/amd64/usr/bin:/home/douglas/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin" \
	TERM="xterm" \
	WINEDEBUG="-all" \
	WINEDLLPATH="/home/douglas/.steam/steam/steamapps/common/Proton 4.2/dist/lib64/wine:/home/douglas/.steam/steam/steamapps/common/Proton 4.2/dist/lib/wine" \
	LD_LIBRARY_PATH="/home/douglas/.steam/steam/steamapps/common/Proton 4.2/dist/lib64:/home/douglas/.steam/steam/steamapps/common/Proton 4.2/dist/lib:/home/douglas/.steam/ubuntu12_32/steam-runtime/pinned_libs_32:/home/douglas/.steam/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib/x86_64-linux-gnu/libfakeroot:/lib/i386-linux-gnu:/usr/lib/i386-linux-gnu:/usr/local/lib:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/lib32:/usr/lib32:/lib:/usr/lib:/home/douglas/.steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu:/home/douglas/.steam/ubuntu12_32/steam-runtime/i386/lib:/home/douglas/.steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu:/home/douglas/.steam/ubuntu12_32/steam-runtime/i386/usr/lib:/home/douglas/.steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu:/home/douglas/.steam/ubuntu12_32/steam-runtime/amd64/lib:/home/douglas/.steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu:/home/douglas/.steam/ubuntu12_32/steam-runtime/amd64/usr/lib:" \
	WINEPREFIX="/home/douglas/.steam/steam/steamapps/compatdata/9420/pfx/" \
	SteamGameId="9420" \
	SteamAppId="9420" \
	WINEDLLOVERRIDES="steam.exe=b;d3d11=n;d3d10=n;d3d10core=n;d3d10_1=n;dxgi=n" \
	STEAM_COMPAT_CLIENT_INSTALL_PATH="/home/douglas/.steam" \
	"/home/douglas/.steam/steam/steamapps/common/Proton 4.2/dist/bin//wine" steam.exe "${@:-${DEF_CMD[@]}}"
```

I've tried adding a  `PROTON_LOG=1 \` line, but that doesn't output the log file either. Though note that other variables seem to work like `GALLIUM_HUD=simple,fps \`

**Comment by pipnina** [comment]
I just ran the game, hosted a lobby with my brother in both 4.2-3 and 3.16-9
In a few seconds the game running under 4.2-3 started to desync while the 3.16-9 did not.
Log files attached
[steam-9420-3.16-9.log](https://github.com/ValveSoftware/Proton/files/3178047/steam-9420-3.16-9.log)
[steam-9420-4.2-3.log](https://github.com/ValveSoftware/Proton/files/3178048/steam-9420-4.2-3.log)

**Comment by tatsujb** [comment]
@pipnina how do you obtain these logs?

**Comment by pipnina** [comment]
I ran it through steam without the FAF launcher. Since the bug is to do with the normal vanilla game it makes sense to troubleshoot the version as packaged by steam.

**Comment by Cyclic3** [comment]
Using the latest 3.16 I keep freezing and getting this whilst playing on a large map for a while
```
173410.403:0008:004d:trace:module:GetModuleFileNameW L"Z:\\games\\fast\\steamapps\\common\\Supreme Commander Forged Alliance\\bin\\SupremeCommander.exe"
173415.409:0008:004d:err:ntdll:RtlpWaitForCriticalSection section 0xfeec00 "?" wait timed out in thread 004d, blocked by 0046, retrying (60 sec)
```

This appears to be some form of deadlock.

I have got this error before in wine with FL studio. Does anyone have any ideas, it's getting rather annoying.

**Comment by tatsujb** [comment]
@kisak-valve first batch of tests under version 4.11-1 of proton seem positive: the desync issue that was present with version 4.2-9 seems to have gone away.

well done :)

**Comment by StringCheesian** [comment]
When I resize the game window, the game's UI doesn't properly resize with it.

Kernel 5.2.4-1-MANJARO, nvidia 430.26, KDE frameworks 5.60.0, KDE plasma 5.16.3
Proton 4.11 (with esync disabled)
GTX 1050 Ti

This is before resizing:
![supcom_before](https://user-images.githubusercontent.com/21348553/62675600-802be180-b95c-11e9-87ec-fc848c2e4ec0.png)

This is after resizing:
![supcom_after](https://user-images.githubusercontent.com/21348553/62675624-96d23880-b95c-11e9-9867-56724b211553.png)

**Comment by tatsujb** [comment]
generally this game under linux runs poorly when you use windowed mode. 

you haven't found this bug yet, I presume, but when you are mid-game and you resize the window you get an unfixable blackscreen. The game didn't crash, it's still technically perfectly playable, you just can't see anything, so realistically absolutely unplayable. 

easy solution is to use fullscreen mode. which is ironic because performance and stability-wise this game runs better in windowed mode under Windows.

**Comment by DarkFenX** [comment]
Debian Testing
Xorg
Nvidia 418.88 + GTX 1080
Gnome DE with 3840x2160 with 200% scaling

On 4.11-3, when I launch sc: fa, i get error "Unable to create Direct3D. Please ensure system has current video drivers." (either game or wine, not native to the system).

**Comment by tatsujb** [comment]
verstion 4.11-10 prevents Forged Alliance from running altogether.

**Comment by varizon** [comment]
Hello. Cyrillic does not work in steam proton. How to add support? Everything works in wine. Tell me please.
https://user-images.githubusercontent.com/11276627/78422126-d2094080-767e-11ea-9ec9-04031c9426a6.jpg

**Comment by tatsujb** [comment]
@kisak-valve 

Can we look into getting this game into the platinum range?

the sim works flawlessly but alot of the FX (like napalm and other 2D projectile sprite) fail to show up.

this of course was the case since the very first version of wine but I expect experimenting with all of the levers we now have, DXVK, d9vk, D3Dwine, gallium, ect, 3d party dependencies, ect could potentially solve the issue.

<details><summary>Screenshots</summary><p>

under windows : 

![image](https://user-images.githubusercontent.com/5132359/93024092-f422c980-f5f3-11ea-9271-0055e4c5c272.png)

under linux :

![Screenshot from 2020-09-13 19-09-18](https://user-images.githubusercontent.com/5132359/93024180-c722e680-f5f4-11ea-8d96-ae1d24a41ae3.png)

![Screenshot from 2020-09-13 19-12-49](https://user-images.githubusercontent.com/5132359/93024371-449b2680-f5f6-11ea-98f4-a4afa19742ed.png)

</p></details>

**Comment by tatsujb** [comment]
@kisak-valve 

I unearthed a new issue.

I couldn't see it before as I was stuck on kernel 5.2 

but now I am on kernel 5.8 and I can see this issue.

On kernel 5.2 I could go all the way up to Proton 5.13 and the call of a generated run file from terminal (PROTON_DUMP_DEBUG_COMMANDS=1)  would work.

but on linux kernel 5.8 I find that running with run file only works up to Proton 4.11-13, after this (and all the way up to proton experimental) it fails with : 
```
wineserver: using server-side synchronization.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
wine: Using RTKit to control niceness in the [-15,15] range
wine: cannot find L"/init"
Setting breakpad minidump AppID = 9420
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198023671480 [API loaded no]
Forged Alliance terminated with exit code 0
```

The differences between the run files are as follows : 

Proton 4.11-13 (the one that works) : 

```
#!/bin/bash
#Run game or given command in environment

cd "/home/t/.local/share/Steam/steamapps/common/Supreme Commander Forged Alliance"
DEF_CMD=("/home/t/.local/share/Steam/steamapps/common/Supreme Commander Forged Alliance/bin/SupremeCommander.exe")
PATH="/home/t/.local/share/Steam/steamapps/common/Proton 4.11/dist/bin/:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/bin:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/bin:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/t/.dotnet/tools" \
	TERM="xterm" \
	WINEDEBUG="-all" \
	WINEDLLPATH="/home/t/.local/share/Steam/steamapps/common/Proton 4.11/dist/lib64//wine:/home/t/.local/share/Steam/steamapps/common/Proton 4.11/dist/lib//wine" \
	LD_LIBRARY_PATH="/home/t/.local/share/Steam/steamapps/common/Proton 4.11/dist/lib64/:/home/t/.local/share/Steam/steamapps/common/Proton 4.11/dist/lib/:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_64:/usr/lib/x86_64-linux-gnu/libfakeroot:/lib/i386-linux-gnu:/usr/local/lib:/lib/x86_64-linux-gnu:/lib:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/lib/x86_64-linux-gnu:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/lib:/home/t/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib:/home/t/.local/share/Steam/steamapps/common/Supreme Commander Forged Alliance" \
	WINEPREFIX="/home/t/.local/share/Steam/steamapps/compatdata/9420/pfx/" \
	SteamGameId="9420" \
	SteamAppId="9420" \
	WINEDLLOVERRIDES="steam.exe=b;mfplay=n;dxvk_config=n;d3d11=n;d3d10=n;d3d10core=n;d3d10_1=n" \
	STEAM_COMPAT_CLIENT_INSTALL_PATH="/home/t/.local/share/Steam" \
	"/home/t/.local/share/Steam/steamapps/common/Proton 4.11/dist/bin/wine" steam.exe "${@:-${DEF_CMD[@]}}"
```

Proton-Experimental (that doesn't work) : 
```
#!/bin/bash
#Run game or given command in environment

cd "/home/t/.local/share/Steam/steamapps/common/Supreme Commander Forged Alliance"
DEF_CMD=("/home/t/.local/share/Steam/steamapps/common/Supreme Commander Forged Alliance/bin/SupremeCommander.exe")
PATH="/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/bin/:/usr/bin:/bin" \
	TERM="xterm" \
	WINEDEBUG="-all" \
	WINEDLLPATH="/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/lib64//wine:/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/lib//wine" \
	LD_LIBRARY_PATH="/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/lib64/:/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/lib/:/overrides/lib/x86_64-linux-gnu:/overrides/lib/i386-linux-gnu" \
	WINEPREFIX="/home/t/.local/share/Steam/steamapps/compatdata/9420/pfx/" \
	SteamGameId="9420" \
	SteamAppId="9420" \
	WINEDLLOVERRIDES="steam.exe=b;dotnetfx35.exe=b;dxvk_config=n;d3d11=n;d3d10=n;d3d10core=n;d3d10_1=n;d3d9=n" \
	STEAM_COMPAT_CLIENT_INSTALL_PATH="/home/t/.local/share/Steam" \
	WINE_LARGE_ADDRESS_AWARE="1" \
	GST_PLUGIN_SYSTEM_PATH_1_0="/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/lib64/gstreamer-1.0:/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/lib/gstreamer-1.0" \
	WINE_GST_REGISTRY_DIR="/home/t/.local/share/Steam/steamapps/compatdata/9420/gstreamer-1.0/" \
	MEDIACONV_AUDIO_DUMP_FILE="/home/t/.local/share/Steam/steamapps/shadercache/9420/fozmediav1/audio.foz" \
	MEDIACONV_AUDIO_TRANSCODED_FILE="/home/t/.local/share/Steam/steamapps/shadercache/9420/swarm/transcoded_audio.foz" \
	MEDIACONV_VIDEO_DUMP_FILE="/home/t/.local/share/Steam/steamapps/shadercache/9420/fozmediav1/video.foz" \
	MEDIACONV_VIDEO_TRANSCODED_FILE="/home/t/.local/share/Steam/steamapps/shadercache/9420/swarm/transcoded_video.foz" \
	"/home/t/.local/share/Steam/steamapps/common/Proton - Experimental/dist/bin/wine" steam.exe "${@:-${DEF_CMD[@]}}"
```

from mixing and matching I've been able to assert that it's not any of the surrounding args changes or additions that constitute this breaking change.

it's probably not even the changes within wine itsel but rather how it is called. 

from the error : `RLIMIT_NICE is <= 20, unable to use setpriority safely` we can deduce this is an issue of Niceness level. 
which may have changed with the new kernel. 

on ubuntu I seem to recall default niceness was 20 but now when I type `ulimit -e` I get 0.

I also notice that in the previous arguments that were called there is a fakeroot: `/usr/lib/x86_64-linux-gnu/libfakeroot:` but not in the versions after Proton 4.11-13.

not that adding that line back helps, perhaps there lies the answer anyways. could you help me get proton-experimental working from terminal for Forged Alliance, please?

I've noticed drastic performance improvements in other games such as rocket league with the new Proton Experimental so I'd like to benefit from this and perhaps the new protons will help bridge the gap in missing VFX in the end.

**Comment by Bottinator22** [comment]
I can play this game just fine with Proton 4.11-13, with esync disabled. Anything newer has issues or does not work at all.

However, some effects are missing. And by that, I mean a LOT of effects are missing, around half of them I believe. Some projectiles are completely invisible or are just thin white lines.

I first played this game on Windows and, for the past few years, have been playing it on Linux. I'm starting to miss these effects when I watch a Windows user play the game...

**Comment by kisak-valve** [comment]
Hello @Bottinator22, please elaborate on what issues this game has when run with newer Proton versions. Also, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression(s), and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Additionally, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by Bottinator22** [comment]
> Hello @Bottinator22, please elaborate on what issues this game has when run with newer Proton versions. Also, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression(s), and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Additionally, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

From what I've experienced before, Proton 5.0-10 causes desyncs in multiplayer games. Might be fixed in FAF, but not sure.
This is the log with 5.13-6:
[steam-9420.log](https://github.com/ValveSoftware/Proton/files/8356238/steam-9420.log)
This is the log with 7.0-1:
[steam-9420.log](https://github.com/ValveSoftware/Proton/files/8356239/steam-9420.log)

https://gist.github.com/Bottinator22/2f9fbcad969b349df8644df8ad2b55e9

**Comment by Bottinator22** [comment]
The logs don't seem to contain anything at all...

Log with 4.11-13, which I am using:
[steam-9420.log](https://github.com/ValveSoftware/Proton/files/8356244/steam-9420.log)

**Comment by kisak-valve** [comment]
That sounds like there's some kind of issue before Proton tries to run the game itself. If you completely close Steam, then run `steam` from a terminal, does it give any hint in the terminal spew when you tell Steam to start the game with newer Proton?

Your system information is missing the extended diagnostics information. Please let Steam gather that information and update your gist.

**Comment by Bottinator22** [comment]
alright

**Comment by Bottinator22** [comment]
https://gist.github.com/Bottinator22/2f9fbcad969b349df8644df8ad2b55e9

**Comment by Bottinator22** [comment]
`chdir /home/roger/.local/share/Steam/steamapps/common/Supreme Commander Forged Alliance`

that's all it logs when I start the game

**Comment by kisak-valve** [comment]
Blind guess, can you verify the integrity of `Steam Linux Runtime - Soldier`'s files?

**Comment by Bottinator22** [comment]
I actually run Proton 3.16 to use FAF
Proton 4.11-13 seems to run the same however

**Comment by Bottinator22** [comment]
> Blind guess, can you verify the integrity of `Steam Linux Runtime - Soldier`'s files?

all files successfully validated

**Comment by Bottinator22** [comment]
Just noticed the game updated through Steam, didn't see any differences tho

**Comment by tatsujb** [comment]
@kisak-valve Hey I've done more testing. 

I've confirmed with a battery of tests that Forged Alliance (9420) doesn't run out of the box on a fresh install of Steam. 

It would run if Rocket League (252950) is installed in proton mode. (didn't need to have run just installed) 

from there I was able to work out that the missing piece of the puzzle was "Steamworks Shared"

with that installed FA  (9420)  does run. (VIDEO  : https://youtu.be/6Tw_kWJzkyo )

I'm trying to find out how to force steam to install it.

Either way I think it should be default, as rare as the use case may be, there will be people out there trying to play 9420 on linux without first installing another proton game requiring  "Steamworks Shared" (is it's APPID : 228980 ?)

also I've been trying to run these types of commands : 

`steam steam://install/228980`

as described here : https://github.com/ValveSoftware/steam-runtime
and here : https://github.com/ValveSoftware/steam-for-linux/issues/3474

but I can't get the command to do anything.

**Comment by tatsujb** [comment]
@kisak-valve Sorry for the low video encode quality, I was in a rush. this video demonstrates the bug (and I think it also demonstrates that "Steamworks Shared" is the solution.) 

https://youtu.be/V3bz4L50Wz0

**Comment by tatsujb** [comment]
@kisak-valve https://youtu.be/6Tw_kWJzkyo

**Comment by tatsujb** [comment]
@kisak-valve hey! please help! do you have an update? have the appropriate people been made aware?

**Comment by tatsujb** [comment]
@kisak-valve can you reproduce?

**Comment by kisak-valve** [comment]
Hello @tatsujb, friendly reminder that I'm a moderator for Valve's issue trackers on Github, and not a Steam or Proton developer myself. We'll need to hear from a Steam dev if they need more help reproducing that snafu.

It should be noted that Steamworks Shared isn't downloaded as one big collection of redistributable installers. There was a bunch of common and duplicated installers which were bundled with many different games. Those got pulled out of the individual games and put in a shared folder which gets managed to match what was originally bundled with the individual games. The question to ponder is why Steam isn't reliably pairing a specific first time installer with the game, instead of generalizing to a download of the Steamworks Shared depot as a whole.

**Comment by tatsujb** [comment]
@kisak-valve yeah I expected that would be the case I just don't have the right jargon and knowledge to express it.

Sorry about mistaking you for a dev. who would be an appropriate person to bring here and look at this issue?

**Comment by tatsujb** [comment]
is anyone going to look into this issue?

**Comment by Bottinator22** [comment]
After updating my own Linux to Kubuntu 22.04, adding dgVoodoo, increasing my Proton version, and... basically, the FAF Linux tutorial, my game works fine, including all effects

**Comment by tatsujb** [comment]
Valve? Hello?

Forged Alliance is still not running when it's the first and only installed game on steam. 

plenty of people on FAF have confirmed to me that they have the same issue.

**Comment by shuusa** [comment]
I've had issues on both Endeavour OS and Gentoo getting this game stable with both LOUD and FAF.
The game runs, and does so fine for a period of time dependent on the general complexity of the game (map size, AI count, etc), but after a while with a match too complex it will freeze, or sometimes with FAF bring up an error dialog.
Usually mentions a memory access violation.

Anyway, I found a workaround for this earlier that somehow got it running on Proton 7.0-4, which fixed the freezing:
All it took was giving Steam a library folder on a separate drive, and moving the game over.
(My second drive is an SSD formatted with BTRFS, in case that matters. Primary Gentoo is EXT4, EOS was BTRFS.)

Not sure _why_ this worked, but if someone else has a similar issue, and you've got another drive with room, give that a try.

**Comment by tatsujb** [comment]
I'm still getting the same issue. I'd like to make an end to end video tutorial for playing FA (9420) with FAF on linux but I can't since I don't know how to provide a universal solution to 9420 not starting if it's the first and only game installed which again is **bound** to happen to some users.

please do something about this : 
![image](https://user-images.githubusercontent.com/5132359/210277713-42ba06c4-40c3-4491-bc83-1dd782d333f6.png)

And, as per usual once the windows version of Rocket League is installed 9420 runs. I haven't been able to find another way to install Steamworks common
![image](https://user-images.githubusercontent.com/5132359/210378017-1cc6a7ae-e610-4dee-b630-3190d8675ada.png)

**Comment by zxq5-dev** [comment]
with the latest version of proton, i can run the game, but the main menus and all the cutscenes completely lag out (including audio during them) everything else works fine once im in game, but anything that isnt a part of the ingame experience such as a cutscene is completely broken / unusable 

im using:
- the latest version of proton
- arch linux
- rx6700xt in case that matters

proton experimental and 6.3 result in the described behaviour, 
proton 4.11 results in a black screen (i changed the esync config like suggested but it still has a black screen)

EDIT: just tried 7.0, same issue

**Comment by sir-ragna** [comment]
@FantasyPvP try putting `PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 PROTON_DEBUG_COMMANDS=1 %command%` in the launch options. `PROTON_NO_ESYNC=1` isn't sufficient anymore you also need the `PROTON_NO_FSYNC=1` option.

A good resource to get games running properly is ProtonDB: https://www.protondb.com/app/9420

**Comment by tatsujb** [comment]
on the latest ubuntu 23.04 + nvidia driver version 530; FA loads up to a black screen. the audio plays, the buttons react (I'm able to navigate by sound) but the screen remains entirely black.

this is both with and without the launch options mentioned above.

also the proton version makes no difference, the issue will persist with Proton-Experimental or the latest  Glorious Eggroll

**Comment by tatsujb** [comment]
> I'm still getting the same issue. I'd like to make an end to end video tutorial for playing FA (9420)...

I've at long last solved the not-first-game-to-be-run issue.

in ubuntu 23 steam://install commands started working for me, so the terminal command :

```
steam steam://install/228980
```
did the trick. 

however on this new version of ubuntu the black screen issue persists. I've tried reinstalling the ubuntu and this time leaving Nvidia graphic drivers untouched (it ships with 525), this did not fix the issue.

**Comment by tatsujb** [comment]
I've got it working now.

I did :

```
sudo apt install protontricks
protontricks 9420 dlls d3dx9 xact
```  

then for the run arguments in steam I put :

```
PROTON_USE_WINED3D=1 PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1 %command% /fullscreen 1920 1080
```

any version of proton should work but I recommend preferring "Experimental"

I've tested the 535 nvidia driver and it works too.

**Comment by tatsujb** [comment]
I've finally updated my video guide

https://youtu.be/4Wm-yZpV_q8

**Comment by fistons** [comment]
There is still not way to make it works on a dual screen setup?

**Comment by Ice-IX** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/688#issuecomment-1573686310

I had the same issue, solved it by setting the resolution manually in Game.prefs to 1920,1080,144 from the default 1024,768,60
Steam has Games.prefs at: [your-steam-path]/steamapps/compatdata/9420/pfx/drive_c/users/steamuser/AppData/Local/Gas Powered Games/Supreme Commander Forged Alliance/Game.prefs
To get the game working with FAF, you will also need to edit the FAF Game.prefs file in its prefix.
I also did not have to fallback to wined3d, DXVK was working fine.

No idea what the actual cause of the problem is, probably a scaling issue with the nvidia driver
